salt/modules/saltcheck.py

```diff
@@ -840,53 +868,123 @@ def _load_file_salt_rendered(self, filepath):
             self.test_dict[key] = value
         return
 
-    def _gather_files(self, filepath):
-        '''
-        Gather files for a test suite
-        '''
-        self.test_files = []
-        filepath = filepath + os.sep + 'saltcheck-tests'
-        for dirname, dummy, filelist in salt.utils.path.os_walk(filepath):
-            for fname in filelist:
-                if fname.endswith('.tst'):
-                    start_path = dirname + os.sep + fname
-                    full_path = os.path.abspath(start_path)
-                    log.info("Found test: %s", full_path)
-                    self.test_files.append(full_path)
-        return
-
-    @staticmethod
-    def _convert_sls_to_path(sls):
-        '''
-        Converting sls to paths
-        '''
-        sls = sls.replace(".", os.sep)
-        return sls
-
     def add_test_files_for_sls(self, sls_name, check_all=False):
         '''
-        Adding test files
+        Detects states used, caches needed files, and adds to test list
         '''
-        sls_split = sls_name.rpartition('.')
-        for path in self.search_paths:
-            if sls_split[0]:
-                base_path = path + os.sep + self._convert_sls_to_path(sls_split[0])
-            else:
-                base_path = path
-            if os.path.isdir(base_path):
-                log.info("searching path: %s", base_path)
-                if check_all:
-                    # Find and run all tests in the state/saltcheck-tests directory
-                    self._gather_files(base_path + os.sep + sls_split[2])
-                    return
-                init_path = base_path + os.sep + sls_split[2] + os.sep + 'saltcheck-tests' + os.sep + 'init.tst'
-                name_path = base_path + os.sep + 'saltcheck-tests' + os.sep + sls_split[2] + '.tst'
-                if os.path.isfile(init_path):
-                    self.test_files.append(init_path)
-                    log.info("Found test init: %s", init_path)
-                if os.path.isfile(name_path):
-                    self.test_files.append(name_path)
-                    log.info("Found test named: %s", name_path)
+        salt_ssh = False
+        if 'running_data/var/run/salt-minion.pid' in __opts__.get('pidfile', False):
```

Comment:
this check might not work across different OS platforms

Reply:
Yep. That line is the only unique info I can find to detect that salt-client was launched from a salt-ssh session. If there's an alternative I'm all ears. While likely not an issue for Windows machines much these days, eventually they may also be ssh managed and this will need to be adjusted.

Reply:
I took a glance at opts as well and couldn't find anything that stands out. @thatch45 any better ideas on how to detect an ssh minion here?

Reply:
Any thoughts team? The requirement is to detect that this is running inside of a salt-ssh session.